kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -253,6 +266,9 @@ public void initiateConnect(HostPort remote, List<FilterAndInvoker> filters) {
         if (logFrames) {
             pipeline.addFirst("frameLogger", new LoggingHandler("io.kroxylicious.proxy.internal.UpstreamFrameLogger"));
         }
+        if (dp.isAuthenticationOffloadEnabled()) {
```

Comment:
I'm slightly puzzled here.  I was expecting to see the filter unconditionally added when the `isAuthenticationOffloadEnabled` is false.

Reply:
Doh, the boolean check is backwards. That's right, I wanted it to always be added for the non-offload case.

> If the case where it is true, we've already sent the response anyway.

For the non-SASL-offloading case we will have buffered the api versions request in the `KafkaProxyFrontendHandler` and eventually when the outbound channel is active we forward the buffered message, by then the filters appear to be installed in the pipeline, there is code checking we are in the CONNECTED state at that point and so should have set up the filters. The integration test I've added shows it working okay on the first request.

> I was think ApiVersion filter was becoming an internal concern (like BrokerAddressFilter). Is my mental model wrong?

You are correct, I've made further changes, removing it from all example configurations and the BuiltinFilterContributor, it should be an internal concern.

> aside: Is a Kafka Client allow to request ApiVersions more than once?

Not sure, though I realized that the `ApiVersionsFilter` wasn't being included in the decode predicate, so subsequent requests on the same connection wouldn't have been decoded.